tests/pytests/unit/grains/test_core.py

```diff
@@ -845,13 +852,14 @@ def test_almalinux_8_os_grains(os_release_dir):
 
 def test_unicode_error():
     raise_unicode_mock = MagicMock(name="raise_unicode_error", side_effect=UnicodeError)
-    with patch("salt.grains.core.hostname"):
-        with patch("socket.getaddrinfo", raise_unicode_mock):
-            ret = salt.grains.core.ip_fqdn()
-            assert ret["fqdn_ip4"] == ret["fqdn_ip6"] == []
+    with patch("salt.grains.core.hostname"), patch(
+        "socket.getaddrinfo", raise_unicode_mock
```

Comment:
Well, I see this has been merged now, but I was going to suggest that this be `patch('socket.getaddrinfo', side_effect=UnicodeError)` :upside_down_face: